docs/src/introduction_to_catalyst/advanced_intro_to_catalyst.md

```diff
@@ -0,0 +1,356 @@
+```
+
+The advantage with the Symbolics for is that we can form algebraic expressions from our symbolic variables. E.g. if we (for some reason) would like to evaluate the expression $2X + p$ across the solution vector, we can do this directly using the Symbolic form:
+```@example advanced_intro_to_catalyst_designations
+sol_symb[2X + p]
+```
+Indeed, we can confirm that this returns the expected output:
+```@example advanced_intro_to_catalyst_designations
+sol_symb[2X + p] == 2*sol[X] .+ 1.0
+```
+
+Symbolic variables-based representation is strictly superior to symbol-based representation. However, it is generally only used in combination with programmatic modelling (as during DSL-based modelling we never actually declare the symbolic variables, and thus cannot use them directly). To use symbolic variables-based representation in combination with the DSL, the [`@unpack` macro can be used](@ref ref).
+
+### [System-based symbolic designation](@id advanced_intro_to_catalyst_designations_sys_symb)
+However Catalyst models are created, their species and parameters are internally stored as symbolic variables. Here, it is possible to directly access these through the model itself. E.g. we can access species `X` through:
+```@example advanced_intro_to_catalyst_designations
+rs_prog.X
+```
+we can confirm that this is the same `X` which was used to create the model originally:
+```@example advanced_intro_to_catalyst_designations
+isequal(X, rs_prog.X)
+```
+Here, we can use the symbolic variables stored within the model to represent our initial conditions and parameter values:
+```@example advanced_intro_to_catalyst_designations
+u0_symb = [rs_prog.X => 1.0]
+ps_symb = [rs_prog.p => 1.0, rs_prog.d => 0.2]
+oprob_symb = ODEProblem(rs_prog, u0_symb, tspan, ps_symb)
+sol_symb = solve(oprob_symb)
+nothing # hide
+```
+or to index our solution:
+```@example advanced_intro_to_catalyst_designations
+sol_symb[rs_prog.X]
+```
+
+Since the models we created programmatically and through the DSL are identical, we can access the symbolic variables stored in the DSL-created model using this approach. This can be used to create algebraic expressions when DSL-based modelling is used (where we normally do not have access to the symbolic variables):
+```@example advanced_intro_to_catalyst_designations
+sol_syms[2rs_dsl.X + rs_dsl.p]
+```
+
+This approach for representing species and parameters is primarily used in combination with [hierarchical or compositional modelling](@ref compositional_modeling) (as it enables us to designate species that have identical names, but are stored in different subsystems). System-based symbolic accessing is discussed in more detail [here](@ref ref).
+
+!!! warn
+    When accessing species and parameters that are stored in a system using the above approach it is important to ensure that the model is *complete*. Model completeness is described [here](@ref ref).
+
+## [Independent variables](@id advanced_intro_to_catalyst_independent_vars)
```

Comment:
We should have a short section on independent variables somewhere (I hadn't really heard of this before MTK started using these), and somewhere here seemed natural. The main purpose of this one is that whenever we mention "independent variable" somewhere, we can link this section, and those who are unfamiliar with the term can read up on what it means.